src/app/test/test.component.ts

```diff
@@ -246,25 +255,46 @@ export class TestComponent implements OnInit {
   updatePath(action: string) {
     let reportIds: string[] = this.helperService.getSelectedIds(this.reports);
     if (reportIds.length > 0) {
-      let path = (document.querySelector('#moveToInput')! as HTMLInputElement).value;
-      if (!path.endsWith('/')) {
-        path = path + '/';
-      }
-      if (!path.startsWith('/')) {
-        path = '/' + path;
-      }
+      let path: string = (document.querySelector('#moveToInput')! as HTMLInputElement).value;
+      path = this.addSlashesToPathEnds(path);
       let map: UpdatePathSettings = { path: path, action: action };
       this.httpService.updatePath(reportIds, this.currentView.storageName, map).subscribe(() => this.loadData(path));
     } else {
       this.toastService.showWarning('No Report Selected!');
     }
   }
 
+  addSlashesToPathEnds(path: string): string {
+    if (path.length > 0 && !path.startsWith('/')) {
+      path = '/' + path;
+    }
+    if (!path.endsWith('/')) {
+      path = path + '/';
+    }
+    return path;
+  }
+
   changeFilter(filter: string): void {
+    filter = this.transformFilter(filter);
+    if (filter === this.testFileTreeComponent.rootFolder.name) {
+      filter = '';
+    }
     this.currentFilter = filter;
-    this.reports.forEach((report) => {
+    for (const report of this.reports) {
       report.checked = this.matches(report);
-    });
+    }
+  }
+
+  transformFilter(value: string): string {
+    if (value !== '' && value !== 'Reports') {
+      if (!value.startsWith('/')) {
+        value = `/${value}`;
+      }
+      if (!value.endsWith('/')) {
+        value = `${value}/`;
```

Comment:
Nice, I didn't know this is possible fixed it.

Reply:
I agree that it looks cleaner, but I think that in terms of readability the template string is better as its also used to prefix the string with a slash. Just to stay consistent 😉

Reply:
I think there's a difference between string concatenation and creating a new string. But in this case, having these values so close together it might look nicer.

Reply:
I've created a thingy for this in our conventions document. @Matthbo do you have any opinion on this?

Reply:
I'd say using `+=` for "string building" is pretty clean and clear. If it were taken out of the current scope or put into a new scope (a string passed through to a different function) then using string interpolation is clearer and it returns a new string instead of doing a string manipulation.

Reply:
So you're saying it looks cleaner but consistency is more important and therefore the template is better in this case?

Reply:
No I'm saying that context should decide what the better option is, if you're making a string and manipulating that in the same function then using `+=` is a great solution. If you get a string passed through as an argument then using string interpolation to return a new string is a better solution instead. So yes using string interpolation here is a good solution

Reply:
I see your point, however on line 291 the string is prefixed with a / and on line 294 the string is suffixed with a /. Because consistency is key, i think its better to do both of them the same way. And because the += is not possible for prefixing, i think the template is the better option.

Reply:
As I said, string interpolation (using template literals / template strings) is a good option because of the before mentioned reasons but I also like your reasoning so let's call this resolved for now